### PR TITLE
erasure: fix erasureCreateFile behavior for empty part

### LINKED ISSUE #4660: erasure: createFile does not return an error if empty part is not allowed
If you create an erasure coded file with 0 content and the allowEmpty flag is false the sever should return an io.EOF but it returns nil (success)

## Expected Behavior
sever should return io.EOF

## Current Behavior
server returns success

## Possible Solution
Fix erasureCreateFile

## Steps to Reproduce (for bugs)
add this to erasure-create_test.go
```
_, size, _, err := erasureCreateFile(disks, "testbucket", "testobject1", bytes.NewReader(data[:0]), false, blockSize, dataBlocks, parityBlocks, defaultBitRotAlgorithm, dataBlocks+1)
if err != io.EOF {
	t.Fatal(err)
}
```

### PR DESCRIPTION
**Pull request by aead**
Fixes #4660

## Description
See issue

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/erasure-createfile.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ func erasureCreateFile(disks []StorageAPI, volume, path string, reader io.Reader
 			return nil, 0, nil, traceError(rErr)
 		}
 		if rErr == io.EOF {
+			if bytesWritten == 0 && !allowEmpty {
```
Comment (harshavardhana) [inline cmd/erasure-createfile.go:49]: Under what scenario it is true that allowEmpty is false?

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/erasure-createfile.go (hunk 1 above)
Comment (aead) [reply cmd/erasure-createfile.go:49]: PutObject: first part is can be empty, all other parts not
Even if higher label may prevent this (?!), I would treat this as a bug.
Found this by adding test cases for erasreCreateFile...

Comment (aead) [reply cmd/erasure-createfile.go:49]: *layer

**Comment by aead** [comment]
Will be fixed for encryption, can never occur for sever (top layer checks)

**Pull request closed without merging**